### DISCUSSION TITLE
bind if condition to ref elements own state

### DISCUSSION THREAD

**Discussion by jimmywarting** [alpinejs/alpine#2951] [1. Help] [answered]
You have a grate example of open/close toggle with x-if logic

but i don't want to have to reinvent the wheel so to speak... there is already a element for this kind of thing...
so i'm just wondering: how can i make the x-if reactive to a referens element own open property?

example:
```html
  <details x-ref="detail">
    <template x-if="$refs.detail.open">
      <div>Contents...</div>
    </template>
  </details>
```
code: https://jsfiddle.net/sb6u7zoy/

**Comment by SimoTod** [comment, marked as answer]
Similar to #2950. Try not to use x-ref but a property. Things outside x-data are not generally reactive
